api/envoy/config/route/v3/route.proto

```diff
@@ -122,17 +122,10 @@ message RouteConfiguration {
   //
   google.protobuf.UInt32Value max_direct_response_body_size_bytes = 11;
 
-  // [#not-implemented-hide:]
-  // A list of plugins and their configurations which may be used by a
-  // :ref:`envoy_v3_api_field_config.route.v3.RouteAction.cluster_specifier_plugin`
-  // within the route. All *extension.name* fields in this list must be unique.
-  repeated ClusterSpecifierPlugin cluster_specifier_plugins = 12;
-}
-
-// Configuration for a cluster specifier plugin.
-message ClusterSpecifierPlugin {
-  // The name of the plugin and its opaque configuration.
-  core.v3.TypedExtensionConfig extension = 1;
+  // A list of cluster providers and their configurations which may be used by a
+  // :ref:`cluster provider name <envoy_v3_api_field_config.route.v3.RouteAction.cluster_provider_name>`
+  // within the route. All *name* fields in this list must be unique.
+  repeated core.v3.TypedExtensionConfig cluster_providers = 12;
```

Comment:
This change is not okay.  This will break a field that is actually being used by gRPC.  Just because a field is not used in Envoy does not mean it's okay to change it.

Please revert these API changes.

Reply:
@markdroth So you implemented a hiden API of Envoy in the gRPC? While I think we can revert these API changes this time around, I don't think this is the right pattern. cc @envoyproxy/api-shepherds

Reply:
I think you may be misunderstanding the context of the xDS API.  While this API did start as part of Envoy, these days it's actually moving toward being a general-purpose data plane API, where Envoy is only one of many clients.  In fact, gRPC has xDS client implementations in 4 different languages.  We do sometimes add new fields to the API that gRPC implements support for before Envoy does; when this happens, the field may still be tagged with a "not-implemented-hide" tag for the benefit of Envoy's auto-generated documentation, but that doesn't mean that it's not used by other clients and is okay to break.

That having been said, it looks to me like the [API versioning policy](https://github.com/envoyproxy/envoy/blob/main/api/API_VERSIONING.md#backwards-compatibility) does incorrectly say that the policy does not apply to fields tagged with "not-implemented-hide".  I think that's out of date and needs to be changed, and I'll put together a PR to fix that.

In any case, speaking as one of the API shepherds, it is definitely not okay to break these fields, since we do absolutely have clients of this API that are using them.  Please revert these changes.

Reply:
API versioning policy being updated in #20347.